mcs/class/corlib/System.IO.IsolatedStorage/IsolatedStorageFile.cs

```diff
@@ -872,7 +872,7 @@ public void MoveDirectory (string sourceDirectoryName, string destinationDirecto
 
 			try {
 				Directory.Move (src_full_path, dest_full_path);
-			} catch (IOException) {
+			} catch (UnauthorizedAccessException) {
```

Comment:
`Directory.Move` docs say that it can throw `IOException` under some circumstances, too.  So we probably want to catch both, actually.

Also do you think it makes sense to set the inner Exception of the IsolatedStorageException?  (what's the coreFX IsolatedStorageFile do?)

Reply:
Yea, CoreFX calls this function, which does use the 2 arg IsolatedStorageException constructor:
https://github.com/dotnet/corefx/blob/1453863bc998b63024987e13b1e7a45659b452de/src/System.IO.IsolatedStorage/src/System/IO/IsolatedStorage/IsolatedStorageFile.cs#L704-L709